From 02c252fe1c840ba30042fb9a005b5e97febb778a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Fri, 9 Feb 2024 14:17:35 +0100
Subject: [PATCH 1140/1695] plat-stm32mp2: add support for secure RTC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Compile the RTC PTA and the RTC driver if the RTC driver for stm32 is
enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4aeae883f564d45823c30eedb0086860d3a382fc plat-stm32mp2: add support for secure RTC]
Change-Id: I9fd74a17f67ebbf3ec2e6dede212f7ac4018fd5a
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/402353
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 27b8dc31f..7d1891621 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -275,6 +275,12 @@ ifeq ($(CFG_STM32_BSEC_PTA),y)
 $(call force,CFG_STM32_BSEC3,y,Mandated by CFG_STM32_BSEC_PTA)
 endif
 
+# Enable RTC
+ifeq ($(CFG_STM32_RTC),y)
+$(call force,CFG_DRIVERS_RTC,y)
+$(call force,CFG_RTC_PTA,y)
+endif
+
 # Enable Early TA NVMEM for provisioning management
 CFG_TA_STM32MP_NVMEM ?= y
 ifeq ($(CFG_TA_STM32MP_NVMEM),y)
-- 
2.47.2

